casanovo/denovo/evaluate.py

```diff
@@ -161,7 +161,12 @@ def aa_match(
     pep_match : bool
         Boolean flag to indicate whether the two peptide sequences fully match.
     """
-    if mode == "best":
+    if peptide1 is None and peptide2 is None:
+        return np.array([], dtype=bool), False
+    elif (peptide1 is None) != (peptide2 is None):
+        peptide = peptide1 if peptide2 is None else peptide2
+        return np.array([False] * len(peptide)), False
```

Comment:
**todo:** This can be slightly more efficient.

Now: You create a list of length 1, then create a new list of `len(pep)` elements, and then convert the list to an array.
Instead: `np.zeros(len(peptide), dtype=bool)` would directly create an array of the right size.